client/gutenberg/extensions/publicize/form.jsx

```diff
@@ -19,83 +20,50 @@ import { withSelect, withDispatch } from '@wordpress/data';
 import PublicizeFormUnwrapped from './form-unwrapped';
 
 const PublicizeForm = compose( [
-	withSelect( select => ( {
-		activeConnections: ! select( 'core/editor' ).getEditedPostAttribute( 'publicize' )
-			? []
-			: select( 'core/editor' ).getEditedPostAttribute( 'publicize' ).connections,
-		shareMessage: ! select( 'core/editor' ).getEditedPostAttribute( 'publicize' )
-			? ''
-			: select( 'core/editor' ).getEditedPostAttribute( 'publicize' ).title,
-	} ) ),
-	withDispatch( ( dispatch, ownProps ) => ( {
-		/**
-		 * Directly sets post's publicize data.
-		 *
-		 * Sets initial values for publicize data this saved with post.
-		 * Input parameters are used as defaults.
-		 * They will be ignored if the 'publicize' field has already been set. This prevents
-		 * user changes from being erased every time pre-publish panel is opened and closed.
-		 *
-		 * @param {string} initTitle             String to share post with
-		 * @param {array}  initActiveConnections Array of connection data
-		 */
-		initializePublicize( initTitle, initActiveConnections ) {
-			const { activeConnections, shareMessage } = ownProps;
-			const newConnections =
-				activeConnections.length > 0 ? activeConnections : initActiveConnections;
-			const newTitle = shareMessage.length > 0 ? shareMessage : initTitle;
-			dispatch( 'core/editor' ).editPost( {
-				publicize: {
-					title: newTitle,
-					connections: newConnections,
-				},
-			} );
-		},
+	withSelect( select => {
+		const meta = select( 'core/editor' ).getEditedPostAttribute( 'meta' );
 
+		return {
+			connections: select( 'core/editor' ).getEditedPostAttribute(
+				'jetpack_publicize_connections'
+			),
+			meta,
+			shareMessage: get( meta, [ 'jetpack_publicize_message' ], '' ),
+		};
+	} ),
+	withDispatch( ( dispatch, { connections, meta } ) => ( {
 		/**
-		 * Update state connection enable/disable state based on checkbox.
+		 * Toggle connection enable/disable state based on checkbox.
 		 *
 		 * Saves enable/disable value to connections property in editor
-		 * in field 'publicize'.
+		 * in field 'jetpack_publicize_connections'.
 		 *
-		 * @param {Object}  options              Top-level options parameter
-		 * @param {string}  options.connectionID ID of the connection being enabled/disabled
-		 * @param {boolean} options.shouldShare  True of connection should be shared to, false otherwise
+		 * @param {number}  id ID of the connection being enabled/disabled
 		 */
-		connectionChange( options ) {
-			const { connectionID, shouldShare } = options;
-			const { activeConnections, shareMessage } = ownProps;
-			// Copy array (simply mutating data would cause the component to not be updated).
-			const newConnections = activeConnections.slice( 0 );
-			newConnections.forEach( c => {
-				if ( c.id === connectionID ) {
-					c.should_share = shouldShare;
-				}
-			} );
+		toggleConnection( id ) {
+			const newConnections = connections.map( c => ( {
```

Comment:
If we're altering all these, let's change `c` to the more explicit `connection`.